welkin/models/encounter.py

```diff
@@ -0,0 +1,185 @@
+        )
+
+    def update(self, **kwargs):
+        return super().put(
+            f"{self._client.instance}/patients/{self.patient_id}/encounters/{self.encounter_id}/comments/{self.id}",
+            kwargs,
+        )
+
+    def delete(self):
+        return super().delete(
+            f"{self._client.instance}/patients/{self.patientId}/encounters/{self.encounterId}/comments/{self.id}"
+        )
+
+
+class Comments(Collection):
+    resource = Comment
+    encounter_id: str = None
+    patient_id: str = None
+
+    def __init__(self, encounter_id=None, patient_id=None):
+        self.encounter_id = encounter_id
+        self.patient_id = patient_id
+
+    def get(
+        self,
+        *args,
+        **kwargs,
+    ):
+
+        return super().get(
+            f"{self._client.instance}/patients/{self.patient_id}/encounters/{self.encounter_id}/comments",
+            *args,
+            **kwargs,
+        )
+
+
+class Disposition(Resource):
+    def get(self):
+        return super().get(
+            f"{self._client.instance}/patients/{self.patient_id}/encounters/{self.encounter_id}/disposition"
+        )
+
+
+class Encounter(Resource):
+    sub_resources = [Comment, Comments, Disposition]
+
+    def create(self):
+        return super().post(
+            f"{self._client.instance}/patients/{self.patient_id}/encounters"
+        )
+
+    def get(self, related_data: bool = False):
+        encounters = "encounters"
+        if related_data:
+            encounters = "full-encounters"
+
+        return super().get(
+            f"{self._client.instance}/patients/{self.patient_id}/{encounters}/{self.id}"
+        )
+
+    def update(self, **kwargs):
+        return super().patch(
+            f"{self._client.instance}/patients/{self.patient_id}/encounters/{self.id}",
+            kwargs,
+        )
+
+    def delete(self):
+        return super().delete(
+            f"{self._client.instance}/patients/{self.patient_id}/encounters/{self.id}"
+        )
+
+    @property
+    def Comment(self):
+        if getattr(self, "patient_id"):
+            patient_id = self.patient_id
+        else:
+            patient_id = self.patientId
+        return self._client.Comment(encounter_id=self.id, patient_id=patient_id)
+
+    @property
+    def Comments(self):
+        if getattr(self, "patient_id"):
+            patient_id = self.patient_id
+        else:
+            patient_id = self.patientId
+        return self._client.Comments(encounter_id=self.id, patient_id=patient_id)
+
+    @property
+    def Disposition(self):
+        if getattr(self, "patient_id"):
+            patient_id = self.patient_id
+        else:
+            patient_id = self.patientId
+        return self._client.Disposition(encounter_id=self.id, patient_id=patient_id)
+
+
+class Encounters(Collection):
+    resource = Encounter
+    patient_id: str = None
+    user_id: str = None
+
+    def __init__(self, patient_id=None, user_id=None):
```

Comment:
I don't think we can add a constructor. If we do, objects can't be serialized to this class from JSON responses.